### PR TITLE
Rename rust-snapshot-hash to rust-nightly-date

### PR DESCRIPTION
**Pull request by SimonSapin**
… and remove mentions of "snapshots".

We now use rust-lang.org nightly builds rather than our own snapshots.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9086)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,11 +71,11 @@ If you're running servo on a guest machine, make sure 3D Acceleration is switche
 
 ## The Rust compiler
 
-Servo's build system automatically downloads a snapshot Rust compiler to build itself.
+Servo's build system automatically downloads a Rust compiler to build itself.
 This is normally a specific revision of Rust upstream, but sometimes has a
 backported patch or two.
-If you'd like to know the snapshot revision of Rust which we use, see
-`rust-snapshot-hash`.
+If you'd like to know which nightly build of Rust which we use, see
```
Comment (nox) [inline README.md]: "of Rust we use", no?

**Comment by nox** [comment]
```
./python/servo/bootstrap_commands.py:131: E127 continuation line over-indented for visual indent
./python/servo/bootstrap_commands.py:148: E127 continuation line over-indented for visual indent
./python/servo/bootstrap_commands.py:180: F841 local variable 'docs_url' is assigned to but never used
./python/servo/bootstrap_commands.py:181: E127 continuation line over-indented for visual indent
./python/servo/bootstrap_commands.py:184: F821 undefined name 'docs_url_url'
./python/servo/command_base.py:131: E127 continuation line over-indented for visual indent
````

**Comment by SimonSapin** [comment]
Fixed tidy errors.

**Comment by larsbergstrom** [comment]
Updated https://github.com/servo/servo/wiki/Updating-the-Rust-compiler-used-by-Servo 

:-)

**Pull request merged**